### ISSUE TITLE
Setting Favicon

### ISSUE THREAD

**Issue by phoebus34** [willfaught/paige#68]
Hello. I'm trying to figure out how to set a custom favicon. It seems from the example website you're supposed to put it in the static directory. Mine looks like this: https://nextcloud.phoebus.club/s/WWnZsBDxMngcycr. I'm aware that putting the favicon images in the /images/ folder probably wouldn't change anything. If I run "hugo server" then I try it on firefox I don't get the icon. If I try it on chromium or brave then I get the icon. When I run hugo and upload the files to my server, then the custom icon doesn't appear even on chromium based browsers. Am I doing something wrong? 

Thank you.

**Comment by willfaught** [comment]
>Hello. I'm trying to figure out how to set a custom favicon. It seems from the example website you're supposed to put it in the static directory.

Yep!

>I'm aware that putting the favicon images in the /images/ folder probably wouldn't change anything.

No. Browsers expect them at the URL path root.

>If I run "hugo server" then I try it on firefox I don't get the icon.

Do you get the default icon that comes with Paige?

You want to override all of these files in your own site: https://github.com/willfaught/paige/tree/master/static

I used https://realfavicongenerator.net to generate the ones for Paige. Try using it on your original icon image.

**Comment by phoebus34** [comment]
Okay, I figured it out. It took a little. I see that you really need to override every file if you want a custom icon. Maybe it'd be handy to include the icon generator in the readme and indicate what settings you used to generate them? Also, if I just copied the images into static then when running hugo I'd get an error saying permission denied. I figured out the way to solve this is by deleting all of the default icons in the paige directory. Only then could I run hugo successfully and it'd copy over the files from static. Any icon which you don't have, the default will come back for that one after running hugo. Which, I think that behavior is a little irritating.

Maybe you did mention all this somewhere and I just missed it. Anyway, thank you for the nice theme.

**Comment by willfaught** [comment]
Thanks for the feedback. I added an explanation in the override section.

**Comment by willfaught** [comment]
>Also, if I just copied the images into static then when running hugo I'd get an error saying permission denied.

I've never seen that before. Could it have been bad OS permissions of the files themselves?

>I figured out the way to solve this is by deleting all of the default icons in the paige directory. 

Do you mean the Paige module? If so, that shouldn't be necessary. You only need to put the override files in the same places in your site, e.g. yoursite/static/favicon.ico.

**Comment by phoebus34** [comment]
Sorry, I meant public not paige. I had to delete all the default icons in the public directory for it to not give me a permission error. I assumed it was OS permissions too, so I tried a few things and none of them worked. It was only when I deleted icons from the public directory that it went away. I only got that error when I had the new favicon that had the same filename of an image that was already in the public directory. It could well possibly be a problem with my operating system, but I'm not sure. I was using Debian 12. I think the way to narrow it down would just be to see if somebody can reproduce the same behavior. (That is, just spinning up a new site with the theme, run hugo at least once, then try putting the favicon images inside of static to see whether it gives a permission error).
